flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractJobReconciler.java

```diff
@@ -106,19 +107,27 @@ protected void reconcileSpecChange(
             // We must record the upgrade mode used to the status later
             currentDeploySpec.getJob().setUpgradeMode(availableUpgradeMode.get());
             cancelJob(resource, availableUpgradeMode.get(), observeConfig);
-            newState = JobState.SUSPENDED;
+            if (desiredJobState == JobState.RUNNING) {
+                ReconciliationUtils.updateStatusBeforeDeploymentAttempt(resource, deployConfig);
```

Comment:
Is this redundant, we do not do upgrade in this branch. This will be done in the `if (currentJobState == JobState.SUSPENDED && desiredJobState == JobState.RUNNING)` I think.